src/components/authentication-error-dialog/index.tsx

```diff
@@ -0,0 +1,97 @@
+import { useCallback, useEffect, useRef } from 'react'
+import { IconSignIn16 } from '@hashicorp/flight-icons/svg-react/sign-in-16'
+import { IconX16 } from '@hashicorp/flight-icons/svg-react/x-16'
+import { AuthErrors } from 'types/auth'
+import { safeAnalyticsTrack } from 'lib/analytics'
+import useAuthentication from 'hooks/use-authentication'
+import Button from 'components/button'
+import Dialog from 'components/dialog'
+import Text from 'components/text'
+import { guessMethodOfDismissal } from './helpers'
+import s from './authentication-error-dialog.module.css'
+
+/**
+ * A dialog that renders for `AuthErrors.RefreshAccessTokenError` errors.
+ * Provides an option for users to sign in again. When dismissed, automatically
+ * signs the user out.
+ */
+const AuthenticationErrorDialog = () => {
+	const closeButtonRef = useRef<HTMLButtonElement>()
+	const { error, signIn, signOut } = useAuthentication()
+	const isRefreshAccessTokenError = error === AuthErrors.RefreshAccessTokenError
+	const label = 'Are you still there?'
+
+	useEffect(() => {
+		if (isRefreshAccessTokenError) {
+			safeAnalyticsTrack('AuthenticationErrorDialog rendered', {
+				error,
+			})
+		}
+	}, [error, isRefreshAccessTokenError])
+
+	/**
+	 * Set up callback for when the dialog is dismissed. Tracks an analytics event
+	 * and signs the user out.
+	 *
+	 * @NOTE The dismissEvent variable can be multiple types of events, but
+	 * Reach's DialogProps only specifies MouseEvent. Probably better to replace
+	 * Reach Dialog with our custom build when the HDS Modal component is stable,
+	 * rather than mess with the TypeScript.
+	 */
+	const handleDismiss = useCallback(
+		// TODO: this can be a click or keydown event
+		(dismissEvent: $TSFixMe) => {
+			const methodOfDismissal = guessMethodOfDismissal({
+				dismissEvent,
+				closeButtonRef,
+			})
+			safeAnalyticsTrack('AuthenticationErrorDialog dismissed', {
+				methodOfDismissal,
+			})
+			signOut()
+		},
+		[signOut]
+	)
+
+	/**
+	 * Set up the callback for the "Sign in" button's click. Tracks an analytics
+	 * event and starts the sign in UI flow.
+	 */
+	const handleButtonClick = useCallback(() => {
+		safeAnalyticsTrack('Click', {
```

Comment:
💭 this will create a new table in DWH with `Click` as the table name. Should we be more descriptive of what we are tracking here?